lib/pluginmanager/update.rb

```diff
@@ -61,7 +65,9 @@ def plugins_to_update(previous_gem_specs_map)
     if update_all?
       previous_gem_specs_map.values.map{|spec| spec.name}
     else
-      not_installed = plugins_arg.select{|plugin| !previous_gem_specs_map.has_key?(plugin.downcase)}
+      # If the plugins isn't available in the gemspec or in 
+      # the gemfile defined with a local path, the plugin is not installed
+      not_installed = plugins_arg.select{|plugin| !previous_gem_specs_map.has_key?(plugin.downcase) && !gemfile.find(plugin) }
```

Comment:
Gemfile#find, only look in the current gemfile it will find normal gem, gems with a path, git or github option.

`LogStash::PluginManager.find_plugins_gem_specs`, look all the available specification.